rl_algorithms/common/helper_functions.py

```diff
@@ -88,3 +88,67 @@ def get_n_step_info(
         next_state, done = (n_s, d) if d else (next_state, done)
 
     return reward, next_state, done
+
+
+def infer_leading_dims(tensor, dim):
+    """Looks for up to two leading dimensions in ``tensor``, before
+    the data dimensions, of which there are assumed to be ``dim`` number.
+    For use at beginning of model's ``forward()`` method, which should
+    finish with ``restore_leading_dims()`` (see that function for help.)
+    Returns:
+    lead_dim: int --number of leading dims found.
+    T: int --size of first leading dim, if two leading dims, o/w 1.
+    B: int --size of first leading dim if one, second leading dim if two, o/w 1.
+    shape: tensor shape after leading dims.
+
+    Cloned from rlpyt repo:
+    https://github.com/astooke/rlpyt/blob/master/rlpyt/models/dqn/atari_r2d1_model.py
+    """
+    lead_dim = tensor.dim() - dim
+    assert lead_dim in (0, 1, 2)
+    if lead_dim == 2:
+        T, B = tensor.shape[:2]
+    else:
+        T = 1
+        B = 1 if lead_dim == 0 else tensor.shape[0]
+    shape = tensor.shape[lead_dim:]
+    return lead_dim, T, B, shape
+
+
+def restore_leading_dims(tensors, lead_dim, T=1, B=1):
+    """Reshapes ``tensors`` (one or `tuple`, `list`) to to have ``lead_dim``
+    leading dimensions, which will become [], [B], or [T,B].  Assumes input
+    tensors already have a leading Batch dimension, which might need to be
+    removed. (Typically the last layer of model will compute with leading
+    batch dimension.)  For use in model ``forward()`` method, so that output
+    dimensions match input dimensions, and the same model can be used for any
+    such case.  Use with outputs from ``infer_leading_dims()``.
+
+    Cloned from rlpyt repo:
+    https://github.com/astooke/rlpyt/blob/master/rlpyt/models/dqn/atari_r2d1_model.py
+    """
+    is_seq = isinstance(tensors, (tuple, list))
+    tensors = tensors if is_seq else (tensors,)
+    if lead_dim == 2:  # (Put T dim.)
+        tensors = tuple(t.view((T, B) + t.shape[1:]) for t in tensors)
+    if lead_dim == 0:  # (Remove B=1 dim.)
+        assert B == 1
+        tensors = tuple(t.squeeze(0) for t in tensors)
+    return tensors if is_seq else tensors[0]
+
+
+def valid_from_done(done):
```

Comment:
It is difficult to understand this function's role though I see this function name and contents. Should modify function or function name.